### ISSUE TITLE
Sometimes, we need to create custom element via js rather than literal markup

### ISSUE THREAD

**Issue by dabeng** [patternfly-webcomponents/patternfly-webcomponents#10]
If we got a function property(**persistentCallbackFn**) for web compoment as shown below, maybe creating custom element via js is better than literal markup because we can encapsulate any complex lines of js code in a manner that is easy to read.
```html
<pf-alert type="danger" persistent="true" persistentCallbackFn="alert('Danger alert closed');">
  <strong>Hey there is a problem!</strong> Yeah this is really messed up and you should <a href="#" class="alert-link">know about it</a>.
</pf-alert>
```

**Comment by priley86** [comment]
I would agree with this. In many scenarios defining a custom element in js is useful (say you want to define a callback method or call a method on initialization like this [plnkr](https://plnkr.co/edit/dL1loV5PgWh4W29Szk1k?p=preview) with React. This is quite common in the Onsen.io examples too. I think this example with the pf-alert is still warranted though (for simple cases).

**Comment by priley86** [comment]
I'll be writing some more examples of this soon and will try to reference this issue.
